spec/controllers/queue_items_controller_spec.rb

```diff
@@ -16,59 +15,136 @@
         expect(flash[:alert]).to_not be_blank
       end
+
+      it 'adds same video to different users queues' do
+        user_2 = Fabricate(:user)
+        post :create, queue_item: {user_id: @user, video_id: video_1}
+        post :create, queue_item: {user_id: user_2, video_id: video_1}
+        expect(user_2.queue_items.count).to eq 1
+      end
+
       it 'redirects to queue_items index page' do
-        post :create, queue_item: {user_id: @user, video_id: video}
+        post :create, queue_item: {user_id: @user, video_id: video_1}
         expect(response).to redirect_to queue_items_path
       end
     end
 
+    describe 'POST update' do
+
+      let(:item_1) { Fabricate(:queue_item, user: @user, video: video_1, order: 1) }
+      let(:item_2) { Fabricate(:queue_item, user: @user, video: video_2, order: 2) }
+
+      context 'valid data' do
+
+        it 'redirects to queue items index page' do
+          post :update_queue, queue_items: [{id: item_1.id, order: 2, rating: 1}, {id: item_2.id, order: 1, rating: 5}]
+          expect(response).to redirect_to my_queue_path
+        end
+
+        it 'puts queue items in order' do
+          post :update_queue, queue_items: [{id: item_1.id, order: 2, rating: 1}, {id: item_2.id, order: 1, rating: 5}]
+          expect(@user.queue_items).to eq([item_2, item_1])
+        end
+
+        it 'starts ordering from 1' do
+          post :update_queue, queue_items: [{id: item_1.id, order: 4, rating: 1}, {id: item_2.id, order: 3, rating: 5}]
+          expect(@user.queue_items.first.order).to eq(1)
+          expect(@user.queue_items.last.order).to eq(2)
+        end
+      end
+
+      context 'invalid data' do
+        it 'redirects to my_queue_path' do
+          post :update_queue, queue_items: [{id: item_1.id, order: 2.5, rating: 1}, {id: item_2.id, order: 1, rating: 5}]
+          expect(response).to redirect_to my_queue_path
+        end
+
+        it 'sets the flash error message' do
+          post :update_queue, queue_items: [{id: item_1.id, order: 2.5, rating: 1}, {id: item_2.id, order: 1, rating: 5}]
+          expect(flash[:error]).to be_present
+        end
+
+        it 'does not change existing queue items' do
+          post :update_queue, queue_items: [{id: item_1.id, order: 3, rating: 1}, {id: item_2.id, order: 2.1, rating: 5}]
+          expect(item_1.reload.order).to eq 1
+        end
+      end
+
+      context 'with other users data' do
+        let(:user_2) { Fabricate(:user) }
+        let(:item_2) { Fabricate(:queue_item, user: user_2, order: 2)}
+        it 'does not change existing queue items' do
+          post :update_queue, queue_items: [{id: item_1.id, order: 3, rating: 1}, {id: item_2.id, order: 1, rating: 5}]
+          expect(item_1.reload.order).to eq 1
+          expect(item_2.reload.order).to eq 2
+        end
+      end
+    end
+
     describe 'DELETE destroy' do
+
       let(:user_2) { Fabricate(:user) }
+      item_1, item_2, item_3 = nil, nil, nil
```

Comment:
This is not a recommended way, creating variables outside of the `let`, `before`, and `it` blocks will be carried to the next spec and possibly influencing it. All these items should be created with `let` or `let!`.